functiondefextractor/core_extractor.py

```diff
@@ -96,123 +200,256 @@ def get_func_body(filename, line_num):
-        final_files = filter_files(test_files)
-    if code.upper() == "FALSE" and test.upper() == "FALSE":
-        final_files = filter_files(get_file_names(path_repo))
-    return final_files
-
-
-def get_test_files(path_repo):
-    """ Function to get test files from the list of all files
-           @parameters
-           path_repo: path to repository
-           @return
-           This function returns the list of test files"""
-    test_files = []
-    for file_paths in get_file_names(path_repo):
-        split_path = file_paths.split(os.sep)
-        for folder_name in split_path:
-            if folder_name in ("test", "tst"):
-                test_files.append(file_paths)
-    return test_files
-
-
 def get_delta_lines(file_name, annot, delta):
     """ Function to get + and - delta number of lines from the annoted method/function
             @parameters
             filename, annot, delta: Path to the file, required annotation, required lines from method """
     line_data = list(filter(None, [line.rstrip() for line in open(file_name)]))
     data = []
     for num, line in enumerate(line_data, 1):
-        if annot.upper() in line.strip().upper():
-            for i in range(0, (int(delta) * 2) + 1):
-                if num - (int(delta) + 1) + i >= len(line_data):
-                    break
-                data.append(line_data[num - (int(delta) + 1) + i])
-            DELTA_BODY.append("\n".join(data))
-            UID_LIST.append(os.path.basename(file_name) + "_")
-            data = []
+        process_delta_lines_body(annot, line, delta, num, line_data, data, file_name)
+        data = []
+
+
+def process_delta_lines_body(annot, line, delta, num, line_data, data, file_name):
+    """ Function to get + and - delta number of lines from the annoted method/function
+        @parameters
+        file_name: Path to the file
+        annot: Required annotation
+        delta: Required lines from method
+        line_data: File content in list format
+        data: variable that holds delta lines data"""
+    if annot.upper() in line.strip().upper():
+        for i in range(0, (int(delta) * 2) + 1):
+            if num - (int(delta) + 1) + i >= len(line_data):
+                break
+            data.append(line_data[num - (int(delta) + 1) + i])
+        DELTA_BODY.append("\n".join(data))
+        UID_LIST.append(os.path.basename(file_name) + "_")
 
 
-def extractor(path_loc, parse_code="True", parse_test="True", annot=None, delta=None):
+def get_flat_list(data_list):
+    """ Function that generates a list by merging a list of sub lists
+        @parameters
+        data_list: list of sub lists
+        @return
+        This function returns a flattened list"""
+    flattened_list = []
+    for val in data_list:
+        if str(type(val)) != "<class 'list'>":
+            flattened_list.append(val)
+        if str(type(val)) == "<class 'list'>":
+            for sub_val in val:
+                flattened_list.append(sub_val)
```

Comment:
Both if check will be checked at each loop, is it necessary ? or an elif can help here?